kyuubi-common/src/main/scala/org/apache/kyuubi/Logging.scala

```diff
@@ -90,32 +91,64 @@ trait Logging {
   }
 
   private def initializeLogging(isInterpreter: Boolean): Unit = {
-    if (Logging.isLog4j12) {
-      val log4j12Initialized = LogManager.getRootLogger.getAllAppenders.hasMoreElements
+    if (Logging.isLog4j2) {
+      val log4j2Initialized = !LogManager.getRootLogger
+        .asInstanceOf[org.apache.logging.log4j.core.Logger].getAppenders.isEmpty
       // scalastyle:off println
-      if (!log4j12Initialized) {
+      if (!log4j2Initialized) {
         Logging.useDefault = true
-        val defaultLogProps = "log4j-defaults.properties"
+        val defaultLogProps = "log4j2-defaults.properties"
         Option(Thread.currentThread().getContextClassLoader.getResource(defaultLogProps)) match {
           case Some(url) =>
-            PropertyConfigurator.configure(url)
+            val context = LogManager.getContext(false).asInstanceOf[LoggerContext]
+            context.setConfigLocation(url.toURI)
           case None =>
             System.err.println(s"Missing $defaultLogProps")
         }
       }
 
       val rootLogger = LogManager.getRootLogger
+        .asInstanceOf[org.apache.logging.log4j.core.Logger]
       if (Logging.defaultRootLevel == null) {
         Logging.defaultRootLevel = rootLogger.getLevel
       }
 
       if (isInterpreter) {
         // set kyuubi ctl log level, default ERROR
         val ctlLogger = LogManager.getLogger(loggerName)
+          .asInstanceOf[org.apache.logging.log4j.core.Logger]
         val ctlLevel = Option(ctlLogger.getLevel()).getOrElse(Level.ERROR)
         rootLogger.setLevel(ctlLevel)
       }
       // scalastyle:on println
+    } else if (Logging.isLog4j12) {
+      val log4j12Initialized =
+        org.apache.log4j.LogManager.getRootLogger.getAllAppenders.hasMoreElements
+      // scalastyle:off println
+      if (!log4j12Initialized) {
+        Logging.useDefault = true
+        val defaultLogProps = "log4j-defaults.properties"
+        Option(Thread.currentThread().getContextClassLoader.getResource(defaultLogProps)) match {
+          case Some(url) =>
+            org.apache.log4j.PropertyConfigurator.configure(url)
+
+          case None =>
+            System.err.println(s"Missing $defaultLogProps")
+        }
+
+        val rootLogger = org.apache.log4j.LogManager.getRootLogger
+        if (Logging.defaultRootLevel == null) {
+          Logging.defaultRootLevel = Level.getLevel(rootLogger.getLevel.toString)
+        }
+
+        if (isInterpreter) {
+          // set kyuubi ctl log level, default ERROR
+          val ctlLogger = org.apache.log4j.LogManager.getLogger(loggerName)
+          val ctlLevel = Option(ctlLogger.getLevel()).getOrElse(org.apache.log4j.Level.ERROR)
```

Comment:
```scala
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/logging/log4j/Level
	at org.apache.kyuubi.Logging.initializeLogging(Logging.scala:141)
	at org.apache.kyuubi.Logging.initializeLoggerIfNecessary(Logging.scala:87)
	at org.apache.kyuubi.Logging.initializeLoggerIfNecessary$(Logging.scala:83)
	at org.apache.kyuubi.engine.spark.SparkSQLEngine$.initializeLoggerIfNecessary(SparkSQLEngine.scala:69)
	at org.apache.kyuubi.Logging.logger(Logging.scala:41)
	at org.apache.kyuubi.Logging.logger$(Logging.scala:39)
	at org.apache.kyuubi.engine.spark.SparkSQLEngine$.logger(SparkSQLEngine.scala:69)
	at org.apache.kyuubi.engine.spark.SparkSQLEngine$.main(SparkSQLEngine.scala:161)
	at org.apache.kyuubi.engine.spark.SparkSQLEngine.main(SparkSQLEngine.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:955)
	at org.apache.spark.deploy.SparkSubmit$$anon$1.run(SparkSubmit.scala:165)
	at org.apache.spark.deploy.SparkSubmit$$anon$1.run(SparkSubmit.scala:163)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1878)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:163)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1043)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1052)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: java.lang.ClassNotFoundException: org.apache.logging.log4j.Level
	at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:419)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
	... 26 more

```

Reply:
can you check this after removing all shaded jars from engine module? the error is caused by our changes

Reply:
checking

Reply:
set defaultLevel type to string